generate_endf80.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import shutil
+import sys
+import tarfile
+import tempfile
+import warnings
+import zipfile
+from multiprocessing import Pool
+from pathlib import Path
+from urllib.parse import urljoin
+
+import openmc.data
+
+from utils import download, process_neutron, process_thermal
+
+# Make sure Python version is sufficient
+assert sys.version_info >= (3, 6), "Python 3.6+ is required"
+
+description = """
+Generate ENDF/B-VIII.0 HDF5 library for use in OpenMC by first processing ENDF
+files using NJOY. The resulting library will contain incident neutron, incident
+photon, and thermal scattering data.
+
+"""
+
+
+class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
+                      argparse.RawDescriptionHelpFormatter):
+    pass
+
+
+parser = argparse.ArgumentParser(
+    description=description,
+    formatter_class=CustomFormatter
+)
+parser.add_argument('-d', '--destination', type=Path, default=Path('endfb80_hdf5'),
+                    help='Directory to create new library in')
+parser.add_argument('--libver', choices=['earliest', 'latest'],
+                    default='earliest', help="Output HDF5 versioning. Use "
+                    "'earliest' for backwards compatibility or 'latest' for "
+                    "performance")
+parser.add_argument('--download', action='store_true',
+                    help='Download zip files from NNDC')
+parser.add_argument('--no-download', dest='download', action='store_false',
+                    help='Do not download zip files from NNDC')
+parser.add_argument('--use-tmpdir', dest='tmpdir', action='store_true',
+                    help='Use temporary directory while processing')
+parser.add_argument('--no-use-tmpdir', dest='tmpdir', action='store_false',
+                    help='Do not use temporary directory while processing')
+parser.add_argument('--extract', action='store_true',
+                    help='Extract zip files')
+parser.add_argument('--no-extract', dest='extract', action='store_false',
+                    help='Do not extract zip files')
+parser.add_argument('--temperatures', type=float,
+                    default=[250.0, 293.6, 600.0, 900.0, 1200.0, 2500.0],
+                    help="Temperatures in Kelvin", nargs='+')
+parser.set_defaults(download=True, extract=True, tmpdir=True)
```

Comment:
I would be tempted to use the same cleanup arguments that we have else where

```suggestion
parser.add_argument('--cleanup', action='store_true',
                    help="Remove download directories when data has "
                    "been processed")
parser.add_argument('--no-cleanup', dest='cleanup', action='store_false',
                    help="Do not remove download directories when data has "
                    "been processed")
parser.set_defaults(download=True, extract=True, cleanup=False)
```